backend/services/implementations/testSessionService.ts

```diff
@@ -163,13 +186,42 @@ class TestSessionService implements ITestSessionService {
     limit?: number,
   ): Promise<Array<TestSessionResponseDTO>> {
     try {
-      const query = MgTestSession.find({ teacher: { $eq: teacherId } });
+      const now = new Date();
+
+      let testSessions: TestSession[];
       if (limit !== undefined) {
-        query.limit(limit);
+        const queries = [
```

Comment:
would aggregation pipeline like this be better? any downsides?
<img width="321" alt="image" src="https://github.com/uwblueprint/jump-math/assets/69697180/0bb69352-b093-4629-aecd-c5fb90ca8a92">

Reply:
Main downside is that the sort you specific isn't valid (same key twice) and to fix it and make the right sort apply to the right objects, we'd have to extract an extra target date field here. I wanted to avoid extracting that field on the backend since it's not super clear what it means (depends on the copy we choose on the frontend), and it would make the pipeline quite large

Reply:
Let me know your thoughts though

Reply:
makes sense, was curious what pipeline would look like but seems too complex